Framework/API/src/ExperimentInfo.cpp

```diff
@@ -1115,17 +1118,11 @@ ExperimentInfo::getInstrumentFilename(const std::string &instrumentName,
 
 /** Return a const reference to the DetectorInfo object.
  *
- * Any modifications of the instrument or instrument parameters will invalidate
+ * Setting a new instrument via ExperimentInfo::setInstrument will invalidate
```

Comment:
You could probably figure out a way to avoid invalid references, but the result would still be nonsense: If you have two threads, where one sets a new instrument (invalidating the reference to `DetectorInfo` in the current implementation), and the other thread loops over all detectors, to e.g., read the position, what should the outcome be?

- The positions of the old instrument?
- The positions of the new instrument?
- Some positions from the old positions, the rest from the new instrument?

What I am saying is that doing such things is a data race with unpredictable outcome (unless you do global locking of all instrument access), so I would rather have crashed from accessing invalid references: that at least tell me that something is wrong.